File: app/src/main/java/com/tournamentbracketjava/Models/BracketActivityModel.java
package com.tournamentbracketjava.Models;

import android.content.Context;
import com.tournamentbracketjava.Contracts.BracketInterface;
import com.google.gson.Gson;
import java.io.IOException;
import java.util.List;
import java.io.InputStream;

public class BracketActivityModel implements BracketInterface.Model{

    private Context context;
    private GridDataClass finalGridsValues;

    public BracketActivityModel(Context context){
        this.context=context;
        String f = readJSONFromAsset();
        Gson gson = new Gson();
        finalGridsValues = gson.fromJson(f, GridDataClass.class);
    }

    @Override
    public int getWinningTeam() {
        int rowSize = finalGridsValues.getC().getFinalsGrids().get(0).getRows().size();
        return finalGridsValues.getC().getFinalsGrids().get(0).getRows().get(rowSize - 1).getItems().get(0).get(0).getWinnerTeamID();
    }

    @Override
    public List<Rows> getRows() {
        return finalGridsValues.getC().getFinalsGrids().get(0).getRows();
    }

    @Override
    public List<Annotations> getAnnotations() {
        return finalGridsValues.getC().getFinalsGrids().get(0).getAnnotations();
    }

    @Override
    public List<Connections> getConnections() {
        return finalGridsValues.getC().getFinalsGrids().get(0).getConnections();
    }

    private String readJSONFromAsset() {
        String json;
        try {
            assert context != null;
            InputStream is = context.getAssets().open("bracket.json");
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        } catch (IOException ex) {
            ex.printStackTrace();
            return null;
        }
        return json;
    }

}

File: app/src/main/res/layout/com/bracket/Models/BracketActivityModel.kt
package com.bracket.Models

import android.content.Context
import com.google.gson.Gson
import java.io.InputStream

class BracketActivityModel(private val context: Context) : BracketInterface.Model{

//    val finalsGrid: FinalsGridEndpoint = FinalsGridApi.getFinalsGrid()!!.create(
//        FinalsGridEndpoint::class.java)
//
//    val call: Call<GridDataClass> = finalsGrid.getFinalGrid()
//
//    val response = call.execute()
//    val finalGridsValues: GridDataClass? = response.body()

    private val f = readJSONFromAsset()
    private val gson: Gson = Gson()
    private val finalGridsValues: GridDataClass = gson.fromJson(f, GridDataClass::class.java)

    override fun getWinningTeam(): Int {
        val rowSize= finalGridsValues.c.finalsGrids[0].rows.size
        return finalGridsValues.c.finalsGrids[0].rows[rowSize-1].items[0][0].winnerTeamID
    }

    override fun getRows(): List<Rows> {
        return finalGridsValues.c.finalsGrids[0].rows
    }

    override fun getAnnotations(): List<Annotations> {
        return finalGridsValues.c.finalsGrids[0].annotations
    }

    override fun getConnections(): List<Connections> {
        return finalGridsValues.c.finalsGrids[0].connections
    }

    private fun readJSONFromAsset(): String? {
        val json: String?
        try {
            val  inputStream: InputStream = context.assets.open("bracket.json")
            json = inputStream.bufferedReader().use{it.readText()}
        } catch (ex: Exception) {
            ex.printStackTrace()
            return null
        }
        return json
    }

}